src/content/datachannel/datatransfer/js/main.js

```diff
@@ -41,97 +43,78 @@ megsToSend.addEventListener('change', function(e) {
   }
 });
 
-function createConnection() {
+async function createConnection() {
   sendButton.disabled = true;
   megsToSend.disabled = true;
+  // Safari needs to trigger the media permission for RTCDataChannel to work
```

Comment:
use adapter.browserDetails.browser. Also s/for RTCDataChannel to work/for the ice connection to work due to ice candidate restrictions/